packages/mui-base/src/useSwitch/useSwitch.ts

```diff
@@ -115,11 +155,24 @@ export default function useSwitch(props: UseSwitchParameters): UseSwitchReturnVa
     onBlur: createHandleBlur(otherProps),
   });
 
+  const getRootProps: UseSwitchReturnValue['getRootProps'] = (otherProps = {}) => ({
+    ref: handleInputRef,
```

Comment:
The same ref is used for both `root` and `input` slots.

Reply:
I'm not sure how to change it. On one hand, the `getRootProps` is a replacement for input on the `Switch`, so it makes sense (to me, at least) that getRootProps gets `handleInputRef`. On the other hand, `Radio` and `Checkbox` don't have a hidden input so they use `getInputProps` for the actual input element. Therefore, `getInputProps` also gets `handleInputRef`

Reply:
So even though same ref is used for `root` and `input`, it is still working correctly. In `Switch`, `handleInputRef` refers to the `Root` but in `Radio` and `Checkbox`, `handleInputRef` refers to the input.